state/state.go

```diff
@@ -1332,6 +1334,24 @@ func (st *State) processIAASModelApplicationArgs(args *AddApplicationArgs) error
 	return nil
 }
 
+func (st *State) processCAASModelApplicationArgs(args *AddApplicationArgs) error {
+	// TODO(caas) - remove once units are supported.
+	args.NumUnits = 0
+
+	if args.Series == "" {
+		// args.Series is not set, so use the series in the URL.
+		args.Series = args.Charm.URL().Series
+		if args.Series == "" {
+			// Should not happen, but just in case.
+			return errors.New("series is empty")
+		}
+	}
+
+	// TODO(caas) restrict the series to CAAS series.
```

Comment:
could we make all of the TODOs "TODO(caas)" so we can easily search

Reply:
done